### PR TITLE
💥 Update ALL remaining roles to new format (with docs and tests)

### PR DESCRIPTION
**Pull request by anarion80**
**What this PR does / why we need it**:

As in the title: I updated all the remaining roles to new format - adding docs and molecule tests. I know this doesn't really adhere to your contributing guideline, but you're probably busy, PR intake is slow, so here it is.

While creating passing tests I had to fix some roles:
- Krusader had old, non-existent image
- Nomad needed gpg key update
- Timemachine needed apt cache update
- Wallabag needed additional directories created with certain permissions (can be a bug in Wallabag currently)
- ZNC needed a starting config file as it was interactive and thus not easy to automate testing for

From the general things:
- Fixed lint
- Moved from deprecated links to networks for certain roles
- Added variables for container image names:versions instead of hardcoding in tasks
- Updated Ansible module names to use fully qualified collection name (FQCN)

All passing tests are here: https://github.com/anarion80/ansible-nas/actions/runs/4392401003

**Which issue (if any) this PR fixes**:

Fixes #

**Any other useful info**:

### CODE REVIEW

**Comment by davestephens** [comment]
Wow, this is amazing, thank you!

I'll start reviewing this week, hopefully won't take too long to get merged.

File: roles/znc/templates/znc.conf.j2
Comment (davestephens) [inline roles/znc/templates/znc.conf.j2:1]: Why is this file changing?

**Review by davestephens — CHANGES_REQUESTED** [review]
This looks great, thank you for taking the time!

Few comments;

- all of the doc moves have been dragged and dropped rather than `git mv ....`, which means the files are showing as new additions, and we lose the history.
- the roles I'd already rejigged are missing the new docker settings - these are a nice addition so would be good to add
- the docs generation doesn't currently cater for docs within the role directories, so will need some shell scripting to set up (or the docs be left in their original location for now).

File: roles/znc/templates/znc.conf.j2
Comment (anarion80) [reply roles/znc/templates/znc.conf.j2:1]: Without the config, ZNC container is restarting:

```
Checking for list of available modules... 
Opening config [/znc-data/configs/znc.conf]... [ No such file ]
Restart ZNC with the --makeconf option if you wish to create this config.
Unrecoverable config error.
```

expecting a run with --makeconf.

Restarting container is not good for automated checking in test, I figured updating tasks to deal with interactive config is also not really the best way to go - hence the ready-made config.

File: roles/emby/molecule/default/verify.yml
[the remark this answers is not in the document]
Comment (anarion80) [reply roles/emby/molecule/default/verify.yml:1]: I think it was already there: https://github.com/davestephens/ansible-nas/blob/master/roles/emby/molecule/default/verify_stopped.yml

File: roles/nomad/molecule/default/verify.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    - name: Get Nomad service state
+      ansible.builtin.systemd:
+        name: nomad
+        state: started
+      register: service_result
+
+    - name: Check that Nomad service is running
+      ansible.builtin.assert:
+        that:
+          - service_result.status['ActiveState'] == "active"
```
[the remark this answers is not in the document]
Comment (anarion80) [reply roles/nomad/molecule/default/verify.yml:19]: Yes, I think so:

```
"name": "nomad",
    "state": "started",
    "status": {
        "ActiveEnterTimestamp": "Fri 2023-03-17 08:20:51 UTC",
        "ActiveEnterTimestampMonotonic": "998571988",
        "ActiveExitTimestamp": "Fri 2023-03-17 08:20:51 UTC",
        "ActiveExitTimestampMonotonic": "998544668",
        "ActiveState": "active",
        ...
        "Result": "success",
        ....
```

But you're probably right that this:
```
- name: Get Nomad service state
      ansible.builtin.systemd:
        name: nomad
        state: started
```

should be rather:
```
- name: Get Nomad service state
      ansible.builtin.systemd:
        name: "{{ nomad_service_name }}"
        state: started
```

**Comment by anarion80** [comment]
> all of the doc moves have been dragged and dropped rather than git mv ...., which means the files are showing as new additions, and we lose the history.

Do you mean moves of existing docs from `website` into roles? If so - yes, I can remove then, do `git mv ...` and update if necessary.

> the roles I'd already rejigged are missing the new docker settings - these are a nice addition so would be good to add

Which setting do you mean exactly?

> the docs generation doesn't currently cater for docs within the role directories, so will need some shell scripting to set up (or the docs be left in their original location for now)

Yes, so the docs are left in their original settings (when I add new roles for myself I add docs to both locations for now), and in new roles location - just to be consistent with `hello_world` example and ready for any future script updates.

**Comment by anarion80** [comment]
Well, I copying the files using git-cp which was supposed to copy git history, but I don't see that git history :/

**Comment by davestephens** [comment]
> Well, I copied the files using git-cp which was supposed to copy git history, but I don't see that git history :/

Ah, that makes sense why they are showing as new files. I think it'll end up as a management nightmare having two copies of the docs - so please can we remove the new copies and leave the docs where they were originally. Once I get to sorting out the docs generation I can move them into the roles again!

**Comment by anarion80** [comment]
Ok, I removed the new/individual docs, and also rebased to upstream

**Comment by davestephens** [comment]
Finally got this one merged - thanks for your efforts - I also did a fair bit of extra work on it :)

Also - the following tests aren't passing:

```
FAILED roles/joomla/molecule/default/molecule.yml::test
FAILED roles/nextcloud/molecule/default/molecule.yml::test
FAILED roles/openhab/molecule/default/molecule.yml::test
FAILED roles/piwigo/molecule/default/molecule.yml::test
FAILED roles/transmission-with-openvpn/molecule/default/molecule.yml::test
FAILED roles/virtual_desktop/molecule/default/molecule.yml::test
```

It seems that they are all roles that have been flipped to creating a network over the legacy `link` method. Did you see this in testing?

**Comment by anarion80** [comment]
Thanks! Yes, I fillpped them over to the new method over deprecated link method.
I quickly checked the first two roles on `new_format` branch and they are passing tests all right.
I will rebase my master and check there.

**Comment by davestephens** [comment]
Interestingly they pass on GitHub, but fail locally. I'll do some more
digging!

**Pull request merged**